ddtrace/internal/processor/trace.py

```diff
@@ -226,3 +227,34 @@ def shutdown(self, timeout):
         except ServiceStatusError:
             # It's possible the writer never got started in the first place :(
             pass
+
+
+@attr.s
+class SingleSpanSamplingProcessor(SpanProcessor):
+    """SpanProcessor for sampling single spans:
+    - Users can specify span rules with service, name, sample rate, and rate limit.
+    - Those rules are used to assess all spans that are not sampled by trace sampling.
+    If the rules assess that a span should be kept, we add specific tags to it.
+    - The single spans that will be kept are sent in separate payloads if stats computation is enabled.
+    If stats computation is not enabled, then the single spans will be sent along with normal trace payloads.
+    """
+
+    rules = attr.ib(type=(List[SpanSamplingRule]))
+
+    def on_span_start(self, span):
+        # type: (Span) -> None
+        pass
+
+    def on_span_finish(self, span):
+        # type: (Span) -> None
+        # only sample if the span isn't already going to be sampled by trace sampler
+        if span.context.sampling_priority is not None and span.context.sampling_priority <= 0:
```

Comment:
what if the sampling priority changes after this span has finished?

Reply:
Hmmmm... under what circumstances could that occur? Shouldn't sampling priority be the same from span start onwards?

Reply:
it can be changed manually via `span.set_tag(MANUAL_KEEP)`, or we might have other products (like ASM) who might decide at some point during the request that it should be kept (or dropped)

Reply:
some of the other language implementations "lock" the sampling decision under some conditions, but we haven't. so technically it is mutable up until the point when we encode it to put onto the wire

Reply:
Hmmmm ok, so do you think we should put this somewhere other than the final span processor? We can maybe move it a bit further along to right before encoding? @majorgreys  what do you think regarding this?

Reply:
can we also be more explicit about what has to be done with `sampling_priority` _before_ this trace processor is applied? Like we need the existing tracer sampling to be applied before this trace processor runs, right?

Probably worth documenting this dependence in the class docstring